cmd/tink-worker/client/tink/tink_test.go

```diff
@@ -0,0 +1,151 @@
+			t.Error(fmt.Errorf("%s.Serve exited with error: %w", name, err))
+		}
+	}()
+
+	dialer := func(ctx context.Context, _ string) (net.Conn, error) {
+		return listener.DialContext(ctx)
+	}
+
+	conn, err := ggrpc.DialContext(ctx, "bufnet", ggrpc.WithContextDialer(dialer), ggrpc.WithInsecure())
+	if err != nil {
+		t.Fatalf("Failed to dial %s: %v", name, err)
+	}
+
+	client := workflow.NewWorkflowServiceClient(conn)
+	return client
+}
+
+func TestObtainServerCredsWithInvalidCerts(t *testing.T) {
+	emptyCert := ""
+	svr := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprint(w, emptyCert)
+	}))
+	defer svr.Close()
+
+	certTests := []struct {
+		testName         string
+		url              string
+		expectedErrorMsg string
+	}{
+		{"certURL arg validation", "", "certURL cannot be empty"},
+		{"ensure we get a response when fetching a cert", "something", "unable to fetch cert from something"},
+		{"ensure an empty cert response is rejected", svr.URL, "unable to parse the cert data"},
+	}
+	for _, tt := range certTests {
+		t.Run(tt.testName, func(t *testing.T) {
+			_, err := ObtainServerCreds(tt.url)
+			if err == nil {
+				t.Error("Error expected but none found")
+			} else if fmt.Sprint(err) != tt.expectedErrorMsg {
+				t.Errorf("Error should be %v, got %v", tt.expectedErrorMsg, err)
+			}
+		})
+	}
+}
+
+func TestObtainServerCredsWithValidCert(t *testing.T) {
+	t.Run("using with a self-signed cert", func(t *testing.T) {
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprint(w, selfSignedCert)
+		}))
+		defer server.Close()
+
+		_, err := ObtainServerCreds(server.URL)
+		if err != nil {
+			t.Errorf("Error obtaining tink server creds: %v", err)
+		}
+	})
+}
+
+func TestEstablishServerConnection(t *testing.T) {
+	t.Run("grpcAuthority arg validation", func(t *testing.T) {
+		expectedErrorMsg := "grpcAuthority cannot be empty"
+		_, err := EstablishServerConnection("", nil)
+		if err == nil {
+			t.Error("Error expected but none found")
+		} else if fmt.Sprint(err) != expectedErrorMsg {
+			t.Errorf("Error should be %v, got %v", expectedErrorMsg, err)
+		}
+	})
+
+	t.Run("attempt to connect to a bogus GRPC server", func(t *testing.T) {
+		expectedErrorMsg := "connect to tinkerbell server"
+		_, err := EstablishServerConnection("bogus", nil)
+		if err == nil {
+			t.Error("Error expected but none found")
+		} else if fmt.Sprint(err) != expectedErrorMsg {
+			t.Errorf("Error should be %v, got %v", expectedErrorMsg, err)
+		}
+	})
+}
+
+func TestClientCreation(t *testing.T) {
+	t.Run("ability to create a WorkflowServiceClient", func(t *testing.T) {
+		name := "tinkWorkflowServer"
+		fakeServer, err := grpc.NewServer(log.Test(t, name), func(s *grpc.Server) {
+			workflow.RegisterWorkflowServiceServer(s.Server(), &fakeServer{
+				data: nil,
+				err:  nil,
+			})
+		})
+		if err != nil {
+			t.Errorf("Error: %s", err)
```

Comment:
Stopping this test's execution at this point with `t.Fatalf` makes more sense, IMO. The documentation for `Errorf`, it states:

```
Errorf is equivalent to Logf followed by Fail.
```

... and `Fail`'s documentation says:

```
Fail marks the function as having failed but continues execution
```